src/view/dataElements/xdmObject.jsx

```diff
@@ -61,19 +77,33 @@ const XdmObject = ({
       };
     });
 
-  const schemaOptions = schemasMeta.map(schemaMeta => {
-    return {
-      value: schemaMeta.$id,
-      label: schemaMeta.title
-    };
-  });
+  const doSearch = useCallback(
+    debounce(({ sandboxName, search }) => {
+      setSchemaSearchInProgress(true);
+      return fetchSchemasMeta({
+        orgId: initInfo.company.orgId,
+        imsAccess: initInfo.tokens.imsAccess,
+        sandboxName,
+        search
+      }).then(response => {
+        if (!response.results.length) {
```

Comment:
We need to take into consideration that a new request may be sent while a previous request is still active. In other words, if there's an active request and a subsequent request is made, we need to either cancel the active request (https://davidwalsh.name/cancel-fetch) or just ignore the response to the active request.

Reply:
I think this is addressed with the `CompletionComboBox`